Source/WebCore/platform/graphics/nicosia/NicosiaAnimation.cpp

```diff
@@ -367,8 +372,45 @@ void Animations::resume()
 
 void Animations::apply(Animation::ApplicationResult& applicationResults, MonotonicTime time)
 {
-    for (auto& animation : m_animations)
-        animation.apply(applicationResults, time);
```

Comment:
You are now iterating the animations once, and divide them into 5 buckets, storing adresses of the objects seperated. Then loop over each of those containers. Since typically you have a few animations only registered, I think you are better off if you just loop once per animation type.

Also will be easier to read.

```
auto applyAnimationType = [&](auto type) {
    for (auto& animatin : m_animations) {
        if (animation.keyframes().property() == type)
            animation.apply(applicationResults, time)
    }
};

applyAnimationType(AnimatedProperty::Translate);
applyAnimationType(AnimatedProperty::Scale);
...
```

Reply:
I was thinking about this approach. But it would mean to iterate all animations five times. Whereas in the current solution, we visit every animation only twice. But I'm okay with either implementation.

Reply:
Also, this exact implementation won't work because as the last step, we want to run all animation types except transformation-like animations without a specific order:
- AnimatedProperty::Opacity,
- AnimatedProperty::BackgroundColor,
- AnimatedProperty::Filter,
- AnimatedProperty::WebkitBackdropFilter

Reply:
That's correct the lambda needs to be generalized for that case. Anyhow, I have no strong opinion to the implementation, at the moment you are allocating five temporary vectors - probably all on stack so either way is ok, unless this shows up in profiling 😏